TrustKit/Pinning/TSKSPKIHashCache.m

```diff
@@ -274,16 +274,16 @@ - (SecKeyRef)copyPublicKeyFromCertificate:(SecCertificateRef)certificate
     }
     
     // Get a public key reference for the certificate from the trust
-    SecTrustResultType result;
-    status = SecTrustEvaluate(trust, &result);
-    if (status != errSecSuccess)
+    NSError *error;
+    SecTrustResultType trustResult = 0;
+    if (!evaluateTrust(trust, &trustResult, &error) && (trustResult != kSecTrustResultRecoverableTrustFailure))
```

Comment:
Add a comment explaining why we need to evaluate the chain

Reply:
Done: 
```
    // The certificate chain must be evaluated first in order to be able
    // to determine which is the leaf certificate of the chain, and only
    // then SecTrustCopyKey can be called
```